crates/shared/src/price_estimation/competition.rs

```diff
@@ -73,10 +73,14 @@ impl PriceEstimating for RacingCompetitionPriceEstimator {
                 // Process estimators within each stage in parallel
                 let mut futures: Vec<_> = stage
                     .iter()
-                    .map(|(_, estimator)| estimator.estimate(query.clone()))
+                    .enumerate()
+                    .map(|(index, (_, estimator))| {
+                        let query = query.clone();
+                        async move { (index, estimator.estimate(query.clone()).await) }.boxed()
+                    })
                     .collect();
                 while !futures.is_empty() {
```

Comment:
After reading your comment, I actually think `stream` might make sense here using `futures::stream::FuturesUnordered`.

This avoids us manually having to manipulate the pending future list.

Reply:
I'm inclined to not go that route.
1. Using streams results in a lot of hairy code (e.g. for some reason I'm running into higher kinded lifetime errors which nobody should have to debug if you are just doing an innocent little change).
2. Then I looked into [`FuturesUnordered`](https://docs.rs/futures/latest/futures/stream/struct.FuturesUnordered.html#) which some tokio maintainer [says](https://news.ycombinator.com/item?id=29911141) has a bunch of footguns.
3. Then I looked into tokio's [`JoinSet`](https://docs.rs/tokio/1.32.0/tokio/task/struct.JoinSet.html) which is supposed address those footguns requires some unintuitive unwraps.

All those solutions don't sound very appealing if the only goal is to avoid manually managing the futures list. And btw the compiler already complains if you don't populate `futures` in the loop so I think the risk of mismanaging the list is minimal.

Reply:
This wasn't a hard suggestion - I think the code is good as is. Just thought it might have been easier (but apparently I was wrong).